crates/driver/src/tests/setup/mod.rs

```diff
@@ -332,17 +360,21 @@ pub fn test_solver() -> Solver {
 }
 
 impl Solver {
-    fn address(&self) -> eth::H160 {
+    pub fn address(&self) -> eth::H160 {
         self.private_key.public_address()
     }
 
-    pub fn name(self, name: &str) -> Self {
+    pub fn set_name(self, name: &str) -> Self {
         Self {
             name: name.to_owned(),
             ..self
         }
     }
 
+    pub fn get_name(&self) -> &str {
```

Comment:
On the first glance it looks like this new function is only needed because the JIT order is not part of the solution so there is still logic required to "assemble" the actual solution which we discussed in person.

Please take a look at the `Setup` struct. It has a list of `orders` which are the orders to be solved and a list of `solutions` which are the solutions returned by the solver. You are introducing `jit_orders` separately when it should fit into the `solutions` field because ultimately a jit order will have to be returned as part of a solution.